proof/proof.go

```diff
@@ -0,0 +1,133 @@
+package proof
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ipfs/go-cid"
+	hamt "github.com/ipfs/go-hamt-ipld"
+	"github.com/opentracing/opentracing-go"
+	"github.com/quorumcontrol/chaintree/nodestore"
+	"github.com/quorumcontrol/messages/v2/build/go/gossip"
+	"github.com/quorumcontrol/tupelo-go-sdk/bls"
+	"github.com/quorumcontrol/tupelo-go-sdk/gossip/hamtwrapper"
+	sigfuncs "github.com/quorumcontrol/tupelo-go-sdk/signatures"
+)
+
+func verifyRoundConfirmation(rootCtx context.Context, conf *gossip.RoundConfirmation, quorumCount uint64, verKeys []*bls.VerKey) (bool, error) {
+	sp, ctx := opentracing.StartSpanFromContext(rootCtx, "verifyRoundConfirmation")
+	defer sp.Finish()
+
+	sig := conf.Signature
+	if uint64(sigfuncs.SignerCount(sig)) < quorumCount {
+		return false, nil
+	}
+
+	err := sigfuncs.RestoreBLSPublicKey(ctx, sig, verKeys)
+	if err != nil {
+		sp.SetTag("error", true)
+		return false, fmt.Errorf("error restoring public key: %v", err)
+	}
+
+	msg := conf.RoundCid
+	isVerified, err := sigfuncs.Valid(ctx, sig, msg, nil)
+	if err != nil {
+		sp.SetTag("error", true)
+		return false, fmt.Errorf("error verifying round signature: %v", err)
+	}
+	sp.SetTag("verified", isVerified)
+	return isVerified, nil
+}
+
+func Verify(rootCtx context.Context, proof *gossip.Proof, quorumCount uint64, verKeys []*bls.VerKey) (bool, error) {
```

Comment:
no test here either? I guess if the client can/send/receive then that's enough testing here.